tachyon/math/base/big_int.h

```diff
@@ -422,11 +422,17 @@ struct ALIGNAS(internal::LimbsAlignment(N)) BigInt {
     return *this;
   }
 
-  constexpr BigInt operator/(const BigInt& other) const { return Div(other); }
+  constexpr std::optional<BigInt> operator/(const BigInt& other) const {
+    const std::optional<BigInt> div = Div(other);
+    if (UNLIKELY(!div)) return std::nullopt;
+    return *div;
```

Comment:
This will perform copy. 

```suggestion
    return div;
```